src/LiveComponent/src/Util/FingerprintCalculator.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Symfony\UX\LiveComponent\Util;
+
+class FingerprintCalculator
+{
+    public function __construct(private string $secret)
+    {
+    }
+
+    public function calculateFingerprint(array $data): string
+    {
+        return base64_encode(hash_hmac('sha256', serialize($data), $this->secret, true));
```

Comment:
This is all internal so we can change later but maybe we should create a `ChecksumCalculator` object that's used for calculating the checksum and the fingerprint?

In a future PR, will you need to verify the fingerprint?

Reply:
I'm going to save for later, but you're right.

Related, it's odd to have to pass the `$secret` into `DehydratedComponent`. Probably there should be a `DehydratedComponentFactory` and some of the `DehydratedComponent` logic (i.e. `public static function createFromCombinedData` and the checksum creation) should be moved there.

Reply:
Yeah, a new `ChecksumCalculator` would contain the secret - once we have this service, it will probably require some refactoring of DehydratedComponent.